packages/protocol/src/SavingsContract.ts

```diff
@@ -33,24 +33,32 @@ export function getOrCreateSavingsContract(
   if (version == 2) {
     let tokenEntity = token.getOrCreate(addr)
     savingsContractEntity.token = tokenEntity.id
-  }
-
-  if (massetAddress != null) {
-    savingsContractEntity.masset = massetAddress.toHexString()
-  } else {
-    let underlying = contract.try_underlying()
-    if (underlying.reverted) {
+    let fraction = contract.try_fraction()
+    if (fraction.reverted) {
       throw new Error('Unable to identify underlying for SavingsContract')
     } else {
-      savingsContractEntity.masset = underlying.value.toHexString()
+      savingsContractEntity.fraction = fraction.value
+    }
+
+    if (massetAddress != null) {
+      savingsContractEntity.masset = massetAddress.toHexString()
+    } else {
+      let underlying = contract.try_underlying()
+      if (underlying.reverted) {
+        throw new Error('Unable to identify underlying for SavingsContract')
+      } else {
+        savingsContractEntity.masset = underlying.value.toHexString()
+      }
     }
   }
 
   savingsContractEntity.automationEnabled = false
   savingsContractEntity.cumulativeDeposited = metrics.getOrCreate(addr, 'cumulativeDeposited').id
   savingsContractEntity.cumulativeWithdrawn = metrics.getOrCreate(addr, 'cumulativeWithdrawn').id
   savingsContractEntity.totalCredits = metrics.getOrCreate(addr, 'totalCredits').id
-  savingsContractEntity.totalSavings = metrics.getOrCreate(addr, 'totalSavings').id
+  savingsContractEntity.totalSavings = (
+    metrics.getOrCreate(addr, 'exchangeRate').id * metrics.getOrCreate(addr, 'totalCredits').id
```

Comment:
These are IDs! This change should be reverted - it should just create a metric with a default value (0), because there are no values here to use.